### PR TITLE
Setup linear elliptic executables

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Add executables for the existing Poisson problems, as is done for some of the evolution problems. Since the Metavariables is not specific to the Poisson system anymore (as of #1671), the system is now also chosen when setting up the executable.

### Types of changes:

- [ ] Bugfix
- [x] New feature
- [ ] Refactor

### Component:

- [x] Code
- [ ] Documentation
- [x] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

File: src/Elliptic/Executables/Linear/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+  DomainCreators
+  Informer
+  IO
+  LinearOperators
+  MathFunctions
+  Utilities
+  ${SPECTRE_LIBRARIES}
+  )
+
+function(add_linear_eliptic_executable
```
Comment (geoffrey4444) [inline src/Elliptic/Executables/Linear/CMakeLists.txt]: elliptic

File: src/Elliptic/Executables/Linear/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+# Poisson system with Lorentzian solution
+add_linear_eliptic_executable(
+  PoissonLorentzian3D
+  Poisson::FirstOrderSystem<3>
+  Poisson::Solutions::Lorentzian<3>
+  Poisson::Solutions::Lorentzian<3>
+  "Poisson;PoissonSolutions"
+  )
+
+# Poisson system with Moustache solution
```
Comment (geoffrey4444) [inline src/Elliptic/Executables/Linear/CMakeLists.txt:38]: Comment that this is only a 1D/2D solution

File: src/Elliptic/Executables/Linear/SolveLinearEllipticProblem.hpp [hunk 3]
```diff
@@ -45,31 +45,33 @@
...
-  using system = Poisson::FirstOrderSystem<Dim>;
   using fluxes_computer_tag =
       elliptic::Tags::FluxesComputer<typename system::fluxes>;
 
-  // The analytic solution and corresponding source to solve the Poisson
-  // equation for
-  using analytic_solution_tag =
-      Tags::AnalyticSolution<Poisson::Solutions::ProductOfSinusoids<Dim>>;
+  // All boundary conditions are analytic solutions right now, so we don't need
+  // to check anything. This will be generalized ASAP to handle analytic data
```
Comment (geoffrey4444) [inline src/Elliptic/Executables/Linear/SolveLinearEllipticProblem.hpp]: Clarify what "anything" is

**Review by geoffrey4444 — CHANGES_REQUESTED** [review]
Just some quick minor things, and I'm ok for you to make those changes and then squash, then I'll approve

**Comment by nilsvu** [comment]
Done @geoffrey4444. @kidder could you review or assign a second reviewer?

**Review by kidder — DISMISSED** [review]

**Comment by kidder** [comment]
@geoffrey4444 please look at
@nilsleiffischer this needs a rebase

**Comment by kidder** [comment]
@geoffrey4444 please look at

**Comment by nilsvu** [comment]
@geoffrey4444 please take a quick look, this PR seems ready to merge. Thanks for reviewing!

**Comment by kidder** [comment]
@geoffrey4444 please look at ASAP

**Comment by kidder** [comment]
@geoffrey4444 this should be quick to look at, please do so ASAP

**Review by geoffrey4444 — APPROVED** [review]

**Review by kidder — APPROVED** [review]

**Comment by kidder** [comment]
Ignoring travis failure

**Pull request merged**